cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/centrality/ClosenessCentrality.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.centrality
+
+import com.twitter.cassovary.graph.{GraphDir, BreadthFirstTraverser, Walk, DirectedGraph}
+import scala.collection.immutable
+
+/**
+ * Calculate the closeness centrality of <code>graph</code>
+ * @param normalize Normalize by the ratio of number of connections to other nodes
+ *                  and the number of nodes in the graph
+ */
+class ClosenessCentrality(graph: DirectedGraph, normalize: Boolean = true) extends AbstractCentrality(graph) {
+
+  override def recalculate(): immutable.IndexedSeq[Double] = {
+    graph foreach { node =>
+      val bfs = new BreadthFirstTraverser(graph, GraphDir.OutDir, Seq(node.id), Walk.Limits())
+      var sum = 0.0
+      var neighborCount = 0.0
+
+      bfs foreach { n =>
```

Comment:
make this functional style.
val (sum, neighborCount) = bfs.foldLeft((0,0)) { case ((partialSum, partialCount),a) => (partialSum+a, partialSum+1) } }